lib/vehicle.rb

```diff
@@ -5,14 +5,21 @@ class Vehicle
               :year,
               :make,
               :model,
-              :engine
+              :engine,
+              :registration_date,
+              :plate_type
+  
+  attr_accessor :registration_date,
+                :plate_type
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registration_date = nil
+    @plate_type = nil
```

Comment:
This works because you have created another method in your facility class that `assigns_plates` which is what updates this. 
Should it be the facility classes job to assign a plate type? Or would it be the vehicle classes job to designated the plate type?? 
If you made it in this class, then I would make it its own method below and not have an instance variable here.

Reply:
I'd have to look at the IP, but in my head it makes sense that a vehicle has a plate type but it is assigned elsewhere, like in the real world.

Reply:
Yeah I think that is a potential possibility. My thought is that a vehicle has a plate_type based on whether it is an electric vehicle, an oldie, or just a regular vehicle and then when it gets registered, it is charged based on its plate_type but your way could be true as well.